.github/workflows/lifecycle-image.yml

```diff
@@ -0,0 +1,30 @@
+name: lifecycle-image
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Get metadata
+        id: get_metadata
+        run: |
+          - echo ::set-output name=LIFECYCLE_VERSION::`echo ${{ github.event.release.tag_name }} | cut -d "v" -f2`
+          - echo ::set-output name=BUILDPACK_API::`cat Makefile | grep BUILDPACK_API?= | cut -d "=" -f2`
+          - echo ::set-output name=PLATFORM_API::`cat Makefile | grep PLATFORM_API?= | cut -d "=" -f2`
+      - name: Format label
+        run: |
+          - echo ::set-output name=LABEL_METADATA::`"{\"lifecycle\":{\"version\":\"${{ steps.get_metadata.outputs.LIFECYCLE_VERSION }}\",\"api\":{\"buildpack\":\"${{ steps.get_metadata.outputs.BUILDPACK_API }}\",\"platform\":\"${{ steps.get_metadata.outputs.PLATFORM_API }}\"}}}"`
+      - name: Publish to Registry
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: buildpacksio/lifecycle
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          dockerfile: .github/workflows/lifecycle-image/Dockerfile
+          buildargs: LIFECYCLE_URI=https://github.com/buildpacks/lifecycle/releases/download/v${{ steps.get_metadata.outputs.LIFECYCLE_VERSION }}/lifecycle-v${{ steps.get_metadata.outputs.LIFECYCLE_VERSION }}+linux.x86-64.tgz, LABEL_METADATA=${{ steps.get_metadata.outputs.LABEL_METADATA }}
```

Comment:
It's annoying to have so many args, but it appears we need to format the label directly in the Dockerfile to avoid (somehow) unescaping the surrounding double quotes and terminating the label at the first comma ¯\\_(ツ)_/¯